app/chat_endpoint_handlers.py

```
@@ -0,0 +1,190 @@
+    return {"message": response}
+
+async def chat_message_internal(chat_id: str, user_id: str, action: Action, background_tasks: BackgroundTasks):
+    long_term_summary = redis.get(f"{user_id}-{chat_id}.long_text_summary") or ""
+    medium_term_summary = redis.get(f"{user_id}-{chat_id}.medium_text_summary") or ""
+    short_term_summary = redis.get(f"{user_id}-{chat_id}.short_text_summary") or ""
+    world = ", ".join(json.loads(redis.get(f"{user_id}-{chat_id}.world") or "[]"))
+    characters = []
+    try:
+        characters = list(mongo[mongodb_name(user_id, chat_id)]["characters"].find())
+    except Exception as e:
+        logger.error(f"Failed to retrieve characters from MongoDB: {e}")
```

Ruff (0.12.2)

Undefined name `logger`

(F821)